From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paskal Sitepu <rmnscnce@ya.ru>
Date: Mon, 18 Sep 2023 04:09:42 +0700
Subject: [PATCH 239/384] crypto: kdf_sp800108: Skip kdf selftest

Will shorten the boot time by approx. 400 miliseconds

Taken from [1]

[1]: https://github.com/clearlinux-pkgs/linux/blob/b9925f121fb99c02bdeff88c5ac7b4bc73cb0d6c/kdf-boottime.patch

Signed-off-by: Paskal Sitepu <rmnscnce@ya.ru>
---
 crypto/kdf_sp800108.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/crypto/kdf_sp800108.c b/crypto/kdf_sp800108.c
index c3f9938e1ad2..d148720cd6c9 100644
--- a/crypto/kdf_sp800108.c
+++ b/crypto/kdf_sp800108.c
@@ -125,6 +125,10 @@ static const struct kdf_testvec kdf_ctr_hmac_sha256_tv_template[] = {
 
 static int __init crypto_kdf108_init(void)
 {
+	/* Skip kdf selftest */
+	return 0;
+	/**/
+
 	int ret;
 
 	if (IS_ENABLED(CONFIG_CRYPTO_MANAGER_DISABLE_TESTS))
-- 
2.41.0

